### DISCUSSION TITLE
TFT35v3 E3 and BTT SKR Mini E3 v2.0 - Driving Me Nuts!

### DISCUSSION THREAD

**Discussion by jaanbrotherns** [teeminus/NoTouchScreenFirmware#49] [Q&A]
I've gone through the entire process several times now, starting from scratch, sanity checking, everything! To no avail, now I'm losing my mind. I'm using the exact test config as I have this exact setup, but can't get the screen to work, not even getting the garbled pixels. So, assumption here is I have my pins wrong - anyone think they can help me out?

Config:
[board_pins]
aliases:
    EXP1_1=PB5,  EXP1_3=PA9,   EXP1_5=PA10, EXP1_7=PB8,  EXP1_9=<GND>,
    EXP1_2=PA15, EXP1_4=<RST>, EXP1_6=PB9,  EXP1_8=PB15, EXP1_10=<5V>,

[display]
lcd_type: emulated_st7920
en_pin: EXP1_7
spi_software_sclk_pin: EXP1_6
spi_software_mosi_pin: EXP1_8
spi_software_miso_pin: PA3
encoder_pins: ^EXP1_5, ^EXP1_3
click_pin: ^!EXP1_2

**Comment by teeminus** [comment]
Check the EXP-Alias definition. The one from here (https://github.com/teeminus/NoTouchScreenFirmware/wiki/Klipper-display-configuration#tft35v3-e3-and-btt-skr-mini-e3-v12) looks different.

**Reply by jaanbrotherns** [reply]
Thanks for the ultra quick reply. I'm on 2.0 no 1.2...

**Reply by teeminus** [reply]
Ah, sorry... :(

Have you tried the original BTT firmware (yet/before)?

**Reply by jaanbrotherns** [reply]
I suppose not, I followed the instructions and flashed with the binary provided in this git. Should I go back to original and give that a whirl? Any ideas where I can find it? All the firmware files on the BTT git are for mainboards, not the screen... I think. Really appreciate the help.

**Reply by teeminus** [reply]
You can find the original firmware here: https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware

I think it's at least worth a try. Maybe there is a Klipper config for your mainboard as well.

Have you checked the EXP definition with your mainboard schematic?

**Reply by jaanbrotherns** [reply]
Yea I triple checked the EXP definition, but I'm curious if my aliases to those are not correct. I don't fully understand the EXP1_7 etc and how it works on that board pin. I get <5V> and <GND> but not the others and all. 

Aside.. what you are saying is I may have a shot at using the original screen firmware rather than what has been built in this repository (probably by you)?

**Reply by teeminus** [reply]
This firmware is basically a very stripped down version of the BTT firmware with some (from my point of view) (general) improvements and a different ST7920 emulator.

I have not used the original firmware since the first version of this firmware since this one works better for me. But if this firmware won't work for your setup, maybe the original firmware does. So I would definitely give it a try.

**Reply by jaanbrotherns** [reply]
flashed back to original and no luck there. Going back to your firmware and trying other pinouts.

**Reply by jaanbrotherns** [reply]
@teeminus DUDE! I got it! I flashed original firmware again and then dug through and found this config listing that worked a charm.

[board_pins]
aliases:
;    # EXP1 header
    EXP1_1=PB5,  EXP1_3=PA9,   EXP1_5=PA10, EXP1_7=PB8,  EXP1_9=<GND>,
    EXP1_2=PA15, EXP1_4=<RST>, EXP1_6=PB9,  EXP1_8=PB15, EXP1_10=<5V>,
    # 
    TFT_TX=PA2,  TFT_RX=PA3

[display]
lcd_type: emulated_st7920
en_pin: EXP1_7
spi_software_sclk_pin: EXP1_6
spi_software_mosi_pin: EXP1_8
spi_software_miso_pin: TFT_RX
encoder_pins: ^EXP1_5, ^EXP1_3
click_pin: ^!EXP1_2

From here: https://github.com/teeminus/NoTouchScreenFirmware/issues/30

The above is working on my BTT E3 V2.0 with TFT35 V3.0 (I have a TFT35 E3 V3.0 that also works). Maybe update that for a working confirmed config on the main page?

Thanks for the all the help!

**Reply by teeminus** [reply]
Glad your screen is finally working :)

What do you mean by updating the main page? Both the TFT35V3 and the TFT35v3 E3 is marked as working already. Or did I misunderstand  the question?

**Reply by jaanbrotherns** [reply]
Yea I'm just saying the above may be an add since I wasn't able to get the screen working based off of any of the listed combinations. Of course its up to you but wanted you to have my working config, etc. Thanks again!

**Reply by teeminus** [reply]
I will add the config to the wiki :)
